programs/drift/src/controller/position.rs

```diff
@@ -79,28 +81,45 @@ pub fn get_position_index(user_positions: &PerpPositions, market_index: u16) ->
 pub struct PositionDelta {
     pub quote_asset_amount: i64,
     pub base_asset_amount: i64,
+    pub remainder_base_asset_amount: Option<i64>,
+}
+
+impl PositionDelta {
+    pub fn get_delta_base_with_remainder_abs(&self) -> DriftResult<i128> {
+        let delta_base_i128 =
+            if let Some(remainder_base_asset_amount) = self.remainder_base_asset_amount {
+                self.base_asset_amount
+                    .safe_add(remainder_base_asset_amount.cast()?)?
+                    .abs()
+                    .cast::<i128>()?
+            } else {
+                self.base_asset_amount.abs().cast::<i128>()?
+            };
+        Ok(delta_base_i128)
+    }
 }
 
 pub fn update_position_and_market(
     position: &mut PerpPosition,
     market: &mut PerpMarket,
     delta: &PositionDelta,
 ) -> DriftResult<i64> {
-    if delta.base_asset_amount == 0 {
+    if delta.base_asset_amount == 0 && delta.remainder_base_asset_amount.unwrap_or(0) == 0 {
```

Comment:
we probably wanna have a test where this is hit? do you know what situtations where delta.base_asset_amount would be zero and the delta.remainder_base_asset_amount would not be zero?

Reply:
if you settle and theres only a remainder base asset amount to settle thats when this gets hit (wasnt happening before so entry calcs were missed for remainder sizes)